rust-runtime/aws-smithy-client/src/lib.rs

```diff
@@ -72,13 +72,27 @@ pub mod conns {
     }
 
     #[cfg(feature = "rustls")]
+    /// Return a default HTTPS connector backed by the `rustls` crate.
+    ///
+    /// It requires a minimum TLS version of 1.2.
+    /// It allows you to connect to both `http` and `https` URLs.
     pub fn https() -> Https {
         HTTPS_NATIVE_ROOTS.clone()
     }
 
     #[cfg(feature = "native-tls")]
+    /// Return a default HTTPS connector backed by the `hyper_tls` crate.
+    ///
+    /// It requires a minimum TLS version of 1.2.
+    /// It allows you to connect to both `http` and `https` URLs.
     pub fn native_tls() -> NativeTls {
-        hyper_tls::HttpsConnector::new()
+        let mut tls = hyper_tls::native_tls::TlsConnector::builder();
+        let tls = tls
+            .min_protocol_version(Some(hyper_tls::native_tls::Protocol::Tlsv12))
+            .build()
+            .unwrap_or_else(|e| panic!("Error while creating TLS connector: {}", e));
+        let http = hyper::client::HttpConnector::new();
+        hyper_tls::HttpsConnector::from((http, tls.into()))
```

Comment:
Is the `.into()` necessary? Isn't `tls` already of type `TlsConnector`?

Reply:
Yes, otherwise you get the following compilation error:
```
error[E0277]: the trait bound `hyper_tls::HttpsConnector<_>: From<(hyper::client::HttpConnector, TlsConnector)>` is not satisfied
  --> aws-smithy-client/src/lib.rs:95:9
   |
95 |         hyper_tls::HttpsConnector::from((http, tls))
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(hyper::client::HttpConnector, TlsConnector)>` is not implemented for `hyper_tls::HttpsConnector<_>`
   |
   = help: the trait `From<(T, tokio_native_tls::TlsConnector)>` is implemented for `hyper_tls::HttpsConnector<T>`
```